TheaterSchedule.DAL/Repositories/ResetCodeRepository.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Linq;
+using TheaterSchedule.DAL.Interfaces;
+using TheaterSchedule.DAL.Models;
+using Entities.Models;
+using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
+
+namespace TheaterSchedule.DAL.Repositories
+{
+    public class ResetCodeRepository : IResetCodeRepository
+    {
+        private TheaterDatabaseContext db;
+
+        public ResetCodeRepository(TheaterDatabaseContext context)
+        {
+            db = context;
+        }
+
+        public void Add(ResetCodeModel resetModel)
+        {
+            db.ResetCode.Add(new ResetCode
+            {
+                Code = resetModel.Code,
+                AccountId = resetModel.AccountId,
+                CreationTime = resetModel.CreationTime
+            });
+        }
+
+        public void Delete(int value)
+        {
+            var resetCode = db.ResetCode.FirstOrDefaultAsync(r => r.Code == value);
+            db.ResetCode.Remove(resetCode.Result);
+        }
+
+        public void DropPassword(Account account)
+        {
+            account.PasswordHash = "";
+        }
+
+        public void ResetPassword(Account account, string password)
+        {
+            account.PasswordHash = password;
+        }
+
+        public async Task<ResetCode> GetByValueAsync(int value)
+        {
+            return await db.ResetCode.FirstOrDefaultAsync(r => r.Code == value);
```

Comment:
be carefull when using First - it can hide errors when two records exists
It's better to use single
do we have unique constraint for field Code?

Reply:
Changed FirstOrDefaultAsync to SingleAsync.